torch/ao/sparsity/sparsifier/weight_norm_sparsifier.py

```diff
@@ -0,0 +1,53 @@
+from functools import reduce
+
+import torch
+import torch.nn.functional as F
+
+from .base_sparsifier import BaseSparsifier
+
+def _flat_idx_to_2d(idx, shape):
+    rows = idx // shape[1]
+    cols = idx % shape[1]
+    return rows, cols
+
+class WeightNormSparsifier(BaseSparsifier):
+    def __init__(self,
+                 sparsity_level=0.5, sparsity_pattern=(1, 4),
+                 zeros_per_block=None):
+        if zeros_per_block is None:
+            zeros_per_block = reduce((lambda x, y: x * y), sparsity_pattern)
+        defaults = {
+            'sparsity_level': sparsity_level,
+            'sparsity_pattern': sparsity_pattern,
+            'zeros_per_block': zeros_per_block
+        }
+        super().__init__(defaults=defaults)
+
+    def update_mask(self, layer, sparsity_level, sparsity_pattern,
+                    zeros_per_block, **kwargs):
+        if zeros_per_block != reduce((lambda x, y: x * y), sparsity_pattern):
+            raise NotImplementedError('Partial block sparsity is not yet there')
+        # TODO: Add support for multiple parametrizations for the same weight
+        mask = layer.parametrizations.weight[0].mask
+        if sparsity_level <= 0:
+            mask.data = torch.ones(layer.weight.shape, device=layer.weight.device)
```

Comment:
Why mask.data instead of mask here?

Reply:
This is a constraint that we must have, otherwise a new variable will be created